src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1536,6 +1536,11 @@ internal bool IsTupleTypeOfCardinality(out int tupleCardinality)
         /// </summary>
         internal abstract NamedTypeSymbol NativeIntegerUnderlyingType { get; }
 
+        /// <summary>
+        /// Returns true if the type is defined in source and contains field initializers.
+        /// </summary>
+        internal virtual bool HasFieldInitializers() => false;
```

Comment:
Feels like this should follow our general pattern of being abstract here, rather than virtual.

Reply:
For example, I don't see any changes to WrappedNameTypeSymbol, which I would expect to see.

Reply:
The method is only valid when called on a definition. Added an assert.

Reply:
Re-opened this for the question about abstract versus virtual.

Reply:
Never mind. I'm slow on the uptake.